src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,141 +233,60 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
     }
 
     /**
-     * Scan the schematic and save it to the disk.
+     * Scan the structure and save it to the disk.
      *
      * @param world Current world.
      * @param from  First corner.
      * @param to    Second corner.
      * @return Message to display to the player.
      */
-    public static String saveSchematic(@Nullable World world, @Nullable BlockPos from, @Nullable BlockPos to)
+    public static String saveStructure(@Nullable World world, @Nullable BlockPos from, @Nullable BlockPos to)
     {
         if (world == null || from == null || to == null)
         {
             throw new IllegalArgumentException("Invalid method call, arguments can't be null. Contact a developer.");
         }
 
-        @NotNull Schematic schematic = scanSchematic(world, from, to);
+        BlockPos blockpos =
+                new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
+        BlockPos blockpos1 =
+                new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
+        BlockPos size = blockpos1.subtract(blockpos).add(1, 1, 1);
 
-        String fileName = LanguageHandler.format("item.scepterSteel.scanFormat", schematic.getType(), System.currentTimeMillis());
-        @NotNull File file = new File(getScanDirectory(world), fileName);
+        WorldServer worldserver = (WorldServer) world;
+        MinecraftServer minecraftserver = world.getMinecraftServer();
+        TemplateManager templatemanager = worldserver.getStructureTemplateManager();
 
-        if (SchematicFormat.writeToFile(file, schematic))
-        {
-            return LanguageHandler.format("item.scepterSteel.scanSuccess", fileName);
-        }
-        return LanguageHandler.format("item.scepterSteel.scanFailure");
-    }
-
-    @NotNull
-    private static Schematic scanSchematic(@NotNull World world, @NotNull BlockPos from, @NotNull BlockPos to)
-    {
-        int minX = Math.min(from.getX(), to.getX());
-        int maxX = Math.max(from.getX(), to.getX());
-        int minY = Math.min(from.getY(), to.getY());
-        int maxY = Math.max(from.getY(), to.getY());
-        int minZ = Math.min(from.getZ(), to.getZ());
-        int maxZ = Math.max(from.getZ(), to.getZ());
-        short width = (short) (Math.abs(maxX - minX) + 1);
-        short height = (short) (Math.abs(maxY - minY) + 1);
-        short length = (short) (Math.abs(maxZ - minZ) + 1);
-
-        @NotNull BlockPos minPos = new BlockPos(minX, minY, minZ);
-
-        @NotNull Schematic schematic = new Schematic(DEFUALT_ICON, width, height, length);
-
-        @NotNull BlockPos.MutableBlockPos offset = new BlockPos.MutableBlockPos(0, 0, 0);
+        File theDir = new File("minecolonies");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'theDir' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)